nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -152,9 +152,11 @@ let
 
       ${lib.optionalString cfg.useBootLoader
       ''
-        # Create a writable copy/snapshot of the boot disk.
-        # A writable boot disk can be booted from automatically.
-        ${qemu}/bin/qemu-img create -f qcow2 -F qcow2 -b ${bootDisk}/disk.img "$TMPDIR/disk.img"
+        if ! test -e "$TMPDIR/disk.img"; then
+          # Create a writable copy/snapshot of the boot disk.
+          # A writable boot disk can be booted from automatically.
+          ${qemu}/bin/qemu-img create -f qcow2 -F qcow2 -b ${bootDisk}/disk.img "$TMPDIR/disk.img"
+        fi
```

Comment:
I could use a hand determining if this breaks anything. It's necessary in order for `grub-reboot` to take effect. I could see it breaking things if there were tests existing that expected this to be fresh state every time. My view is, if there are not such tests, this makes it more consistent with the rest of the script (which uses `! test -e` to choose whether to create the other files).

I would like to know if we can throw this at a CI somewhere and see if it breaks any tests, alternatively, if there is a set of tests I can reasonably run locally which would be considered good enough.

Reply:
In case anyone is wondering how it works for the installer and hibernate tests and why I shouldn't do the same here: those tests do a complete install of nixos, and involves a fair amount of extra logic that I would prefer to avoid.

Reply:
I really don't know if this is something we want to do, by default or not at all.
To play safe I would add a `persistBootLoader` option to toggle the behavior and use only where it's needed.

Reply:
Done.